### PR TITLE
[Content collections] `z.object` wrapper for schemas

### PR DESCRIPTION
**Pull request by bholmesdev**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Add `z.object` wrapper around schema definitions. This is now a required wrapper!

### CODE REVIEW

**Review by Yan-Thomas — APPROVED** [review]
LGTM!

File: src/pages/en/guides/content-collections.mdx [hunk 1]
```diff
@@ -144,18 +144,18 @@ You can specify each expected property in the `schema` field of `defineCollectio
 import { z, defineCollection } from 'astro:content';
```
Comment (delucis) [inline src/pages/en/guides/content-collections.mdx:144]: Line 140 above might need updating too?

> You can specify each expected property in the `schema` field of `defineCollection`:

Could become something like?

> You can specify the expected shape of your frontmatter with the `schema` field of `defineCollection`:

File: src/pages/en/guides/content-collections.mdx [hunk 2]
```diff
@@ -189,7 +189,7 @@ defineCollection({
     footnote: z.string().optional(),
     author: z.string().default('Anonymous'),
     language: z.enum(['en', 'es']),
-  }
+  })
 })
 ```
```
Comment (delucis) [inline src/pages/en/guides/content-collections.mdx:195]: We should probably also update the code sample in the “Advanced schema features” section below. (Thanks again for not allowing to comment there GitHub.)

Either wrap in `z.object` or maybe just remove the open/close braces?

```ts
// Allow only strings representing email addresses
authorContact: z.string().email(),
// Allow URL strings only (e.g. `https://example.com/`)
canonicalURL: z.string().url(),
// Parse publishDate as a browser-standard `Date` object
publishDate: z.string().transform(str => new Date(str)),
```

**Review by delucis — COMMENTED** [review]
Thanks Ben! Left a couple of nits but otherwise LGTM. (Helpfully my nits are in the wrong places thanks to GitHub’s UI.)

File: src/pages/en/guides/content-collections.mdx (hunk 2 above)
Comment (bholmesdev) [reply src/pages/en/guides/content-collections.mdx:195]: Ah good catch! And yeah, GitHub reviews have been wacked-out lately. I'll fix

File: src/pages/en/guides/content-collections.mdx (hunk 1 above)
Comment (bholmesdev) [reply src/pages/en/guides/content-collections.mdx:144]: I like it!

**Review by sarah11918 — APPROVED** [review]
LGTM, Ben! Leaving to Chris for the final say!

**Review by delucis — APPROVED** [review]
That’s a `wrap`!

**Pull request merged**